### PR TITLE
[ADP-3161] Add executable for fine-types with conversion to json schema

### PR DESCRIPTION
**Pull request by paolino**
- [x] added an executable that takes only the command convert
- [x] add support for json and yaml as output format for json schema
- [x] report parsing errors in human readable format

### CODE REVIEW

File: lib/fine-types/app/Options.hs [hunk 1]
```diff
@@ -0,0 +1,76 @@
+{-# LANGUAGE ApplicativeDo #-}
```
Comment (HeinrichApfelmus) [inline lib/fine-types/app/Options.hs:1]: As you know, I'm fairly conservative as far as language extensions are concerned, but for this particular use case, I think that's worth it. 🤔

File: lib/fine-types/app/Options.hs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    commands =
+        hsubparser
+            $ mconcat
+                [ command "convert" $ info (Convert <$> convertOptions) mempty
+                -- other commands go here
+                ]
+
+    convertOptions = do
+        input <- strArgument $ metavar "FILE" <> help "input file"
+        output <- strArgument $ metavar "FILE" <> help "output file"
```
Comment (HeinrichApfelmus) [inline lib/fine-types/app/Options.hs:62]: It's a bit bike shedding, but I believe that the tool would be slightly less error-prone to use if one (or even both) of the file arguments are behind a flag, e.g. `-o` for the output file. 🤔 

Having to specify `-o` is the convention for compilers like GHC.
The convention to specify `-i` is used for [`ffmpeg`](https://ffmpeg.org/ffmpeg.html#Synopsis). 🙈

File: lib/fine-types/app/Options.hs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        format <-
+            option formatReader
+                $ long "format"
+                    <> short 'f'
```
Comment (HeinrichApfelmus) [inline lib/fine-types/app/Options.hs:66]:
```suggestion
```

I have tried to find common conventions for command line arguments, but they are a bit sparse:

* [Linux Documentation Project: Standard Command-Line Options](https://tldp.org/LDP/abs/html/standard-options.html)
* [Eric S. Raymond: Command-Line Options](http://catb.org/~esr/writings/taoup/html/ch10s05.html)

In these documents, the short form `-f` is mentioned, but for a different effect like `file` or `force`. The long option is fine, but I would like to disallow the short form here.

File: lib/fine-types/fine-types.cabal [hunk 4]
```diff
@@ -110,3 +110,22 @@ test-suite unit
...
+    , base
+    , bytestring
+    , megaparsec
+    , optparse-applicative
+    , fine-types
+    , yaml
+  other-modules:
+    Options
+
+```
```
Comment (HeinrichApfelmus) [inline lib/fine-types/fine-types.cabal:131]:
```suggestion
```

Stray backticks?

File: schema.json
Comment (HeinrichApfelmus) [inline schema.json:1]: This file should not be in the commit?

File: schema.yaml
Comment (HeinrichApfelmus) [inline schema.yaml:1]: This file should not be in the commit?

**Review by HeinrichApfelmus — APPROVED** [review]
Thank you! 😊 A couple of small issues.

I think that `-o` or `-i` and avoiding the short form `-f` would be beneficial, but I leave a decision on this up to you.

File: lib/fine-types/app/Options.hs (hunk 2 above)
Comment (paolino) [reply lib/fine-types/app/Options.hs:62]: my first version was like that

File: schema.json
Comment (HeinrichApfelmus) [inline schema.json:1]: Stray file?

**Review by HeinrichApfelmus — APPROVED** [review]
Excellent, thank you! 😊

**Pull request merged**